### PR TITLE
PWGGA/GammaConv: Added new cuts for exotic cluster rejection & added JJ weights

### PR DESCRIPTION
**Pull request by jokonig**
-New Cuts for exotic Cluster rejection to check if all cluster cells are only in one T-Card of the EMCal (derived from AliAnalysisTaskEMCALPhotonIsolation)
-Added cuts to test new exotic cluster rejection in Merged EMCal AddTask
- Added Cuts in CaloCalibration task for electron-cluster energy calibration
- Added JJ weighs to r electron-cluster energy calibration histograms

### CODE REVIEW

File: PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx [hunk 1]
```diff
@@ -159,6 +159,7 @@ AliCaloPhotonCuts::AliCaloPhotonCuts(Int_t isMC, const char *name,const char *ti
   fUseElectronClusterCalibration(0),
   fExtendedMatchAndQA(0),
   fExoticEnergyFracCluster(0),
+  fExoticMinEnergyTCard(0),
```
Comment (dmuhlhei) [inline PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx]: needs to be added to the header file

File: PWGGA/GammaConvBase/AliCaloPhotonCuts.h [hunk 2]
```diff
@@ -667,7 +669,7 @@ class AliCaloPhotonCuts : public AliAnalysisCuts {
     TH2F*     fHistTrueNoElectronPositronClusterMatch;  // True No Electron/Positron P vs cluster E in case of matching with a cluster
     TH2F*     fHistElectronClusterMatchTruePID;         // MC true histogram for purity studies of selected electrons
 
-    // histogram for conv candidate rejection 
+    // histogram for conv candidate rejection
     TH2F*     fHistInvMassDiCluster;                    // histogram for monitoring di-cluster mass
     TH2F*     fHistInvMassConvFlagging;                 // histogram for monitoring rejected di-cluster mass
     Int_t     fNMaxDCalModules;                         // max number of DCal Modules
```
Comment (dmuhlhei) [inline PWGGA/GammaConvBase/AliCaloPhotonCuts.h]: classdef increment missing

File: PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx [hunk 3]
```diff
@@ -8190,7 +8231,14 @@ Bool_t AliCaloPhotonCuts::IsExoticCluster( AliVCluster *cluster, AliVEvent *even
 
   if (1-eCross/ecell1 > fExoticEnergyFracCluster) {
     return kTRUE;
-  }  else {
+  } else if ( fUseExoticCluster == 2 && cluster->E() > fExoticMinEnergyTCard ){
```
Comment (dmuhlhei) [inline PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx]: use brackets () behind &&

File: PWGGA/GammaConvBase/AliCaloPhotonCuts.h (hunk 2 above)
Comment (jokonig) [reply PWGGA/GammaConvBase/AliCaloPhotonCuts.h]: added it

File: PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx (hunk 1 above)
Comment (jokonig) [reply PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx]: isnt it in the .h file in Line 516?

Comment (dmuhlhei) [reply PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx]: yes, somehow overlooked it...

File: PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx (hunk 3 above)
Comment (jokonig) [reply PWGGA/GammaConvBase/AliCaloPhotonCuts.cxx]: I added the brackets

**Comment by dmuhlhei** [comment]
+1

**Pull request merged**